eth_account/signers/ledger.py

```diff
@@ -0,0 +1,196 @@
+
+        References:
+        - https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+        - https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+        '''
+        assert(path.startswith('m/'))
+
+        elements = path.split('/')[1:]
+        depth = len(elements)
+
+        result = bytes([depth])  # Number of BIP 32 derivations to perform (max 10)
+
+        for derivation_index in elements:
+            # For each derivation index in the path check if it is hardened
+            hardened = "'" in derivation_index
+            index = int(derivation_index.strip("'"))
+
+            if hardened:
+                # See bip32 spec for hardened derivation spec
+                index = 0x80000000 | index
+
+            # Append index to result as a big-endian (>) unsigned int (I)
+            result += struct.pack('>I', index)
+
+        return result
+
+    def _send_to_device(self, apdu):
+        if self.device is None:
+            self.device = ledgerblue.comm.getDongle(debug=False)
+
+        reply = self.device.exchange(apdu, timeout=60)
+
+        return reply
+
+    @property
+    def address(self):
+        return self.get_address(self.account_id)
+
+    def get_address(self, account_id):
+        '''
+        Query the ledger device for an ethereum address in HD wallet.
+        Offset is the number in the HD wallet tree
+        '''
+        bip32_path = self._path_to_bytes(self.path_prefix + str(account_id))
+
+        # https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc#get-eth-public-address
+        apdu = bytes.fromhex('e0020000')
+        apdu += bytes([len(bip32_path)])
+        apdu += bip32_path
+
+        result = self._send_to_device(apdu)
+
+        # Parse result
+        offset = 1 + result[0]
+        address = result[offset + 1: offset + 1 + result[offset]]
+        address = address.decode()  # Use decode() to convert from bytearray
+
+        return eth_utils.to_normalized_address(address)
+
+    def get_account_id(self, address, search_limit=20, search_account_id=0):
+        '''
+        Convert an address to the HD wallet tree account_id
+        Start search at an account_id. This would allow to search deeper if required.
+        Default search_limit at 20 take about 5s to reach.
+        '''
+        address = eth_utils.to_normalized_address(address)
+
+        for account_id in itertools.count(start=search_account_id):
+            if account_id > search_limit:
+                raise Exception(f'Address {address} not found' +
+                                f'(search_limit={search_limit}, ' +
+                                f'search_account_id={search_account_id})')
+            if address == self.get_address(account_id):
+                return account_id
+
+    def list(self, limit=5, page=0):
+        '''
+        List Ethereum HD wallet adrress of the ledger device
+        '''
+        return list(map(lambda account_id: self.get_address(account_id),
```

Comment:
I think the  lambda isn't needed here, you should be able to just pass in `self.get_address`.

Alternate way to do this which I think is more readable

```python
return [self.get_address(account_id) for account_id in range(...)]
```